wolfCrypt ECC FP cleanup for thread local storage cases

dgarske opened the pull request:
* Added call to `wc_ecc_fp_free()` in wolfCrypt test/benchmark to resolve leak when ECC operations happen from new thread, such as what happens when building with `--enable-stacksize`.
* Added missing `wolfCrypt_Init`to API docs.

Noticed leak with FP_ECC (`--enable-fpecc`) when profiling memory with `--enable-stacksize --enable-trackmemory`. Found variables allocated with THREAD_LS not being free'd in `wolfCrypt_Cleanup` if ECC operations run in a different thread.

toddouska commented on doc/dox_comments/header_files/wc_port.h:
```diff
@@ -17,5 +41,30 @@
+/*!
+    \ingroup wolfCrypt
+    
+    \brief Used to clean up thread specific resources used by wolfCrypt.
```
Need a better description about cases when needed.

toddouska commented on wolfcrypt/src/ecc.c:
```diff
@@ -5923,8 +5923,9 @@ typedef struct {
-   int        lock;            /* flag to indicate cache eviction */
+   word16     lock:1;          /* flag to indicate cache eviction */
                                /* permitted (0) or not (1) */
+   word16     muInit:1;        /* flag to indicate mu was initialized */
```
No protection needed?

toddouska commented on wolfcrypt/src/ecc.c:
```diff
@@ -6466,6 +6467,30 @@ static const struct {
 };
 
 
+static void wc_ecc_fp_free_idx(unsigned x)
```
Please comment function.

toddouska commented on wolfcrypt/src/ecc.c:
```diff
@@ -6520,40 +6538,38 @@ static int find_base(ecc_point* g)
-      wc_ecc_del_point(fp_cache[idx].g);
-      fp_cache[idx].g = NULL;
-      return GEN_MEM_ERR;
+      ret = GEN_MEM_ERR;
```
Why aren't we deling point anymore?

toddouska commented on wolfcrypt/src/ecc.c:
```diff
@@ -6520,40 +6538,38 @@ static int find_base(ecc_point* g)
-         fp_cache[idx].g         = NULL;
-         fp_cache[idx].lru_count = 0;
-         return GEN_MEM_ERR;
+         ret = GEN_MEM_ERR;
```
This is not the same, looks like we're missing code now.

toddouska commented on wolfcrypt/src/ecc.c:
```diff
@@ -7347,22 +7336,24 @@ void wc_ecc_fp_free(void)
-       wc_ecc_fp_free_cache();
+       unsigned x;
+       for (x = 0; x < FP_ENTRIES; x++) {
+          wc_ecc_fp_free_idx(x);
```
Did you try this change with and w/o thread local storage primitives?

dgarske commented:
@toddouska : I reverted the ECC FP cleanup refactor to eliminate possibly introduction of issues and changed to call the `wc_ecc_fp_free();` directly instead of adding a new API.
